timesketch/api/v1/resources/event.py

```diff
@@ -1208,3 +1248,156 @@ def post(self, sketch_id):
+
+        Raises:
+            HTTP_STATUS_CODE_NOT_FOUND: if sketch or event does not exist
+            HTTP_STATUS_CODE_BAD_REQUEST: if the request is malformed, e.g.
+                events or tags are not lists or the number of events or tags
+                exceeds the maximum allowed
+            HTTP_STATUS_CODE_FORBIDDEN: if the user does not have write access
+                to the sketch
+        """
+
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID.")
+
+        if not sketch.has_permission(current_user, "write"):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                "User does not have write access controls on sketch.",
+            )
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        events = form.get("events", [])
+        if not isinstance(events, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Events need to be a list")
+
+        if len(events) > self.MAX_EVENTS_TO_TAG:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} events in a single "
+                "request".format(self.MAX_EVENTS_TO_TAG),
+            )
+
+        searchindex_id = form.get("searchindex_id")
+        searchindex_name = form.get("searchindex_name")
+
+        if not (searchindex_id or searchindex_name):
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No search index information supplied.",
+            )
+
+        searchindex = None
+        # in both cases we are flexible, no matter what was supplied
+        if searchindex_name:
+            searchindex = SearchIndex.query.filter_by(
+                index_name=searchindex_name
+            ).first()
+        elif searchindex_id:
+            searchindex = SearchIndex.query.get(searchindex_id)
+
+        if not searchindex:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to find the Search index.",
+            )
+
+        if searchindex.get_status.status == "deleted":
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to query event on a closed search index.",
+            )
+
+        tags = form.get("tags", [])
+        if not isinstance(tags, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Tags need to be a list")
+
+        if len(tags) > self.MAX_TAGS_PER_REQUEST:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} tags in a single "
+                "request".format(self.MAX_TAGS_PER_REQUEST),
+            )
+
+        datastore = self.datastore
+
+        for _event in events:
+            result = self.datastore.get_event(searchindex.index_name, _event.get("_id"))
+            if not result:
+                logger.warning(
```

Comment:
This warning will not be surfaced to the end user in the frontend. If an event is provided that does not exist, I would recommend aborting instead of a warning.

Reply:
ok my idea was to still continue with the provided data, but also happy to just err out here - will change it

Reply:
ok cool, no aborting then, I will also not add a warning as it might clutter the logs quite a bit.

Reply:
just to be clear here, that would also mean that if a user provided a event id that does not exist, nothing would happen